### DISCUSSION TITLE
Recursive relations

### DISCUSSION THREAD

**Discussion by denny-bee** [laravel/framework#53329] [Ideas]
Currently to retrieve recursive relations, we need to implement an relation which chains a "with" with its own name.
This is not the best way. It calls many SQL queries.
Recently I saw a blog post and video: limits inside eagerloading using CTE.
Could a recursive CTE be the solution for this case.

**Comment by staudenmeir** [comment]
Hi @denny-bee,
I created a package that provides recursive relations using CTEs:
https://github.com/staudenmeir/laravel-adjacency-list

**Reply by denny-bee** [reply]
I know :)
The eager limiting was also a package which they integrated into the framework.
Hoped that they would integrate yours like they did with the other mentioned case.
I think many old-school SQL developers and many others don't know about CTEs.
But they are a powerful tool, to solve many problems.
